src/nbl/ui/CWindowXCB.cpp

```diff
@@ -0,0 +1,296 @@
+    m_dispatcher(*this) {
+    
+    auto& xcb = m_windowManager->getXcbFunctionTable();
+    auto& xcbIccm = m_windowManager->getXcbIcccmFunctionTable();
+
+    m_xcbWindow = xcb.pxcb_generate_id(m_connection->getRawConnection());
+
+    const auto* primaryScreen = m_connection->primaryScreen();
+
+    uint32_t eventMask = XCB_CW_BACK_PIXEL | XCB_CW_EVENT_MASK;
+    uint32_t valueList[] = {
+        primaryScreen->black_pixel,
+        XCB_EVENT_MASK_STRUCTURE_NOTIFY | XCB_EVENT_MASK_KEY_PRESS | XCB_EVENT_MASK_KEY_RELEASE |
+            XCB_EVENT_MASK_FOCUS_CHANGE | XCB_EVENT_MASK_PROPERTY_CHANGE
+    };
+
+    xcb_void_cookie_t xcbCheckResult = xcb.pxcb_create_window(
+        m_connection->getRawConnection(), XCB_COPY_FROM_PARENT, m_xcbWindow, primaryScreen->root,
+        static_cast<int16_t>(m_x),
+        static_cast<int16_t>(m_y),
+        static_cast<int16_t>(m_width),
+        static_cast<int16_t>(m_height), 4,
+        XCB_WINDOW_CLASS_INPUT_OUTPUT, primaryScreen->root_visual, eventMask,
+        valueList);
+
+    setWindowSize_impl(m_width, m_height);
+    
+    auto WM_DELETE_WINDOW = m_connection->resolveAtom(m_WM_DELETE_WINDOW);
+    auto NET_WM_PING = m_connection->resolveAtom(m_NET_WM_PING);
+    auto WM_PROTOCOLS = m_connection->resolveAtom(m_WM_PROTOCOLS);
+    
+    const std::array atoms {WM_DELETE_WINDOW, NET_WM_PING};
+    xcb.pxcb_change_property(
+            m_connection->getRawConnection(), 
+            XCB_PROP_MODE_REPLACE, 
+            m_xcbWindow, 
+            WM_PROTOCOLS, XCB_ATOM_ATOM, 32, atoms.size(), atoms.data());
+
+
+    auto motifHints = fetchMotifMWHints(getFlags().value);
+    m_connection->setMotifWmHints(m_xcbWindow, motifHints);
+    
+    if(isAlwaysOnTop()) {
+        XCBConnection::XCBAtomToken<core::StringLiteral("NET_WM_STATE_ABOVE")> NET_WM_STATE_ABOVE;
+        m_connection->setNetMWState(
+            primaryScreen->root,
+            m_xcbWindow, false, m_connection->resolveAtom(NET_WM_STATE_ABOVE));
+    }
+
+    xcb.pxcb_map_window(m_connection->getRawConnection(), m_xcbWindow);
+    xcb.pxcb_flush(m_connection->getRawConnection());
+    m_dispatcher.start();
+}
+
+CWindowXCB::~CWindowXCB()
+{
+}
+
+IClipboardManager* CWindowXCB::getClipboardManager() {
+    return m_clipboardManager.get();
+}
+
+ICursorControl* CWindowXCB::getCursorControl() {
+    return m_cursorControl.get();
+}
+
+IWindowManager* CWindowXCB::getManager() {
+    return m_windowManager.get();
+}
+
+bool CWindowXCB::setWindowSize_impl(uint32_t width, uint32_t height) {
+    auto& xcb = m_windowManager->getXcbFunctionTable();
+    auto& xcbIccm = m_windowManager->getXcbIcccmFunctionTable();
+
+    xcb_size_hints_t hints = {0};
+
+    xcbIccm.pxcb_icccm_size_hints_set_size(&hints, true, width, height);
+    if(!isResizable()) {
+        xcbIccm.pxcb_icccm_size_hints_set_min_size(&hints, width, height);
+        xcbIccm.pxcb_icccm_size_hints_set_max_size(&hints, width, height);
+    }
```

Comment:
if you look at `IWindow::setWindowSize` we shouldn't be able to get here if the window is not resizable